Adding troubleshooting section to help restoring clusters that were accidently deleted

lucasvmiguel opened the pull request:
I believe a section to explain how to recover a cluster that was accidentally deleted could be beneficial as I see many people having issues like that.

Reference: https://github.com/cloudnative-pg/cloudnative-pg/discussions/5253#discussioncomment-12651012

lucasvmiguel committed:
Adding troubleshooting section to help restoring clusters that were accidently deleted

Signed-off-by: Lucas Vieira <lucas.vieira@road.io>

lucasvmiguel committed:
fix(docs): correct wording in troubleshooting section for cluster restoration

Signed-off-by: Lucas Vieira <lucas.vieira@road.io>

adrianmarcu18 commented:
@lucasvmiguel I think the documentation should be a little bit more clear in terms of naming convention. 
I have tested several times and looks like if the volumesnapshot's source PVC has a different name than what the CNPG expects, the PVC does not get created, and the recovery will fail.

Let's say I want to recover cluster "myapp-postgres". I than create a PVC with the name "myapp-postgres" (without the index number), I do a volume snapshot, than I use that volume snapshot to recover the cluster, CNPG will ignore it and does not  create a properly named PVC.

Additionally, from my tests I see that this step is not needed.

> 5. Delete the recently created PVC so CNPG Operator can recreate a new PVC from the `VolumeSnapshot`.

Following the exact steps from your docs, but skipping this one, the CNPG cluster will be able to use it directly, without having to create it from the VolumeSnapshot. 

Why this is important? Because, if someone wants to automate this process in any way (e.g. build the logic within the CNPG Helm Chart), the VolumeSnapshot could be automatically created for recovery process, without having to delete in advance the existing PVC.

lucasvmiguel committed:
fix(docs): update cluster restoration steps in troubleshooting section

Clarified the process for restoring a deleted cluster by adding a step to save the `claimRef` name and adjusting the order of instructions for creating a new PVC and handling the `VolumeSnapshot`.

Signed-off-by: Lucas Vieira <lucas.vieira@road.io>
